Subject: rt: btrfs locking fix/workaround
From: Ingo Molnar <mingo@elte.hu>
Date: Sun Feb 08 16:59:53 CET 2009

- btrfs_path_lock_waiting() looks rather dubious and there's no
  spin_is_contended() method on PREEMPT_RT - so exclude this for now

=> needs a proper fix later. Either this code gets zapped from btrfs
   upstream, or we add spin_is_contended() to PREEMPT_RT too.

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 fs/btrfs/locking.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6-tip/fs/btrfs/locking.c
===================================================================
--- linux-2.6-tip.orig/fs/btrfs/locking.c
+++ linux-2.6-tip/fs/btrfs/locking.c
@@ -93,6 +93,7 @@ static int btrfs_spin_on_block(struct ex
  */
 int btrfs_try_spin_lock(struct extent_buffer *eb)
 {
+#ifndef CONFIG_PREEMPT_RT
 	int i;
 
 	spin_nested(eb);
@@ -110,6 +111,7 @@ int btrfs_try_spin_lock(struct extent_bu
 			return 1;
 		spin_unlock(&eb->lock);
 	}
+#endif
 	return 0;
 }
 
